### PR TITLE
[WIP]Commenting UI

### PR DESCRIPTION
**Pull request by karamba228**
Added to this PR:
- Added to UI
    - Added current opened file tracking
    - Added "Back" and "New Thread" buttons to control comment threads
    - Added "show resolved" control functionality
    - Added thread header with a profile image, name and the date the thread was posted on
    - Added functional expand button
    - Added thread comments
    - Added collapsed indicator for large threads

Current Functionality:
- Creating new comment threads (works with backend)
- Replying to a comment thread (works with backend)
- Current opened file tracking
- Thread expansion

Needs to be integrated with a backend server:
- Resolving / Reopening a comment thread
- "Sort by" functionality

### CODE REVIEW

File: src/AppHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+          }
+        >
+          {sortItems[key].name}
+        </a>
+      );
+    }
+    return table;
+  }
+
+  setSortState(state?: any, name?: string) {
```
Comment (saulshanabrook) [inline src/AppHeader.tsx]: What if we had one key in the state for what it's sorted by? Like `sorted: 'latest' | 'date' ...`

File: src/CommentFooter.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
+import * as React from 'react';
+
+interface ICommentFooter {
+  expanded?: boolean;
+  handleExpand?: any;
+  replyActive?: boolean;
+  handleReplyActive?: any;
```
Comment (saulshanabrook) [inline src/CommentFooter.tsx]: Could we add type signatures to these? Are they functions? https://www.typescriptlang.org/docs/handbook/functions.html#writing-the-function-type

**Review by saulshanabrook — APPROVED** [review]
This looks good! I left a few minor comments.

File: src/AppHeader.tsx (hunk 1 above)
Comment (karamba228) [reply src/AppHeader.tsx]: I can do that.

Comment (saulshanabrook) [reply src/AppHeader.tsx]: Thank you!

File: src/CommentFooter.tsx (hunk 2 above)
Comment (hoo761) [reply src/CommentFooter.tsx]: I added the signature "VoidFunction" and it works still, is that type correct?

File: src/AppHeader.tsx (hunk 1 above)
Comment (karamba228) [reply src/AppHeader.tsx]: I think i fixed it @saulshanabrook

**Pull request merged**